drivers/ethernet/eth_smsc911x.c

```diff
@@ -531,71 +532,53 @@ static inline void smsc_wait_discard_pkt(void)
 	}
 }
 
-static void smsc_read_rx_fifo(u8_t *buf, u32_t len)
+static void smsc_read_rx_fifo(struct net_pkt *pkt, u32_t len)
 {
-	u32_t *buf32;
+	u32_t buf32;
 
-	__ASSERT_NO_MSG(((uintptr_t)buf & 3) == 0);
 	__ASSERT_NO_MSG((len & 3) == 0 && len >= 4);
 
-	buf32 = (u32_t *)buf;
 	len /= 4;
+
 	do {
-		*buf32++ = SMSC9220->RX_DATA_PORT;
+		buf32 = SMSC9220->RX_DATA_PORT;
+
+		if (net_pkt_write_new(pkt, &buf32, sizeof(u32_t))) {
```

Comment:
@tbursztyka, I'm sorry, but this is terribly slow to call a function for each 4 bytes of a packet. We need to get direct access to net buffer memory back.

The eth_smsc911x.c driver is all broken after your changes. There used to be CI tests for it, but those were removed by @jukkar without any extra thought. Just like you in this PR removed 350 lines of other tests. @nashif, here're good outcomes of the discussion at the network forum last week, where we talked about the need to write tests - I week after, a great number of tests is REMOVED.

Reply:
Various sample applications were removed because of net_app removal. Unfortunately this removed some of the compilation tests at the same time. We should probably have had the same compilation tests in socket version of echo-server/client.

Reply:
@pfalcon it's not like I did not ask you to look at it 6 days ago.

That said, we really need the overlays/config for legacy echo_server/client apps back yes. (we have the same issue for netusb drivers for instance).

Reply:
> @pfalcon it's not like I did not ask you to look at it 6 days ago.

And well, that's what I did, except that I have only 1 head and 2 hands, and I already find that I spend 50% of time in reviews normally, and 75% of time during release rush like we have, but I also need to do some things myself. And surely, I suggested to not haste with net_app removal, and outright warned against removal of 300 lines of tests in all these "release rush" PRs.

> That said, we really need the overlays/config for legacy echo_server/client apps back yes. (we have the same issue for netusb drivers for instance).

There're now https://github.com/zephyrproject-rtos/zephyr/issues/13481 and https://github.com/zephyrproject-rtos/zephyr/issues/13484 for both of these issues.